### DISCUSSION TITLE
neovim lsp could not find symbol with 'import "tensor.flowcore.framework" could not be resolved

### DISCUSSION THREAD

**Discussion by weiyshay** [microsoft/pyright#4860] [Q&A]
Hi,
This is likely a configuration issue and thank you for your feedback.

I am using neovim to working on a python project which uses a custom tensorflow version while the reference is
not addessed with 'import "tensor.flowcore.framework" could not be resolved' like hint.

Is there anyway to tell pyright to reference the code? I have put the tensorflow code to my project dir.

My env:
```
  installed version 1.1.300
  homepage          https://github.com/microsoft/pyright
  languages         Python
  categories        LSP
  executables       pyright-langserver
                    pyright

  → LSP server configuration schema (press enter to expand)
```

**Comment by erictraut** [comment]
Are you saying that the import cannot be "resolved" (i.e. the corresponding module cannot be located)? Or are you saying that the import is being marked as "not referenced" (i.e. pyright thinks that you are importing it but then not referring to the module anywhere within your code)?

If it's an import resolution issue, you can look at pyright's log output for more details. If you include `verboseOutput = true` in the configuration file, it will output even more details to help you diagnose the configuration issue.

**Reply by weiyshay** [reply]
> Are you saying that the import cannot be "resolved" (i.e. the corresponding module cannot be located)? Or are you saying that the import is being marked as "not referenced" (i.e. pyright thinks that you are importing it but then not referring to the module anywhere within your code)?

Eric,
Thank you very much for your response! It is the latter, the code runs ok while I am not able to jump to the relevant source like view c/c++ project. Any tips?

**Reply by erictraut** [reply]
Are you referencing the import? If not, then pyright is doing the correct thing by indicating that you are not referencing the import anywhere in your code.

I'm not sure what you mean by "jump to the relevant source". Are you invoking the "Go to definition" or "Go to declaration" LSP command? If the import is resolved to a ".py" or ".pyi" file, then you should be able to use these commands to open the associated file. If it resolves to a native library (e.g. a ".so" or ".pyd" file), you will not be able to open it in a text editor.

Perhaps you could provide more detailed repro steps? Which library are you trying to import? I don't recognize "tensor.flowcore.framework".

**Reply by hmc-cs-mdrissi** [reply]
I assume they mean [tensorflow.core.framework](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/core/framework). The repository doesn't show it but that folder when installed does have some .py files generated from protobuf files there.

Tensorflow is very much untyped library though. You can pip install types-tensorflow to get some stubs. They are maintained here in [typeshed](https://github.com/python/typeshed/tree/main/stubs/tensorflow/tensorflow/core/framework) and I think tensorflow/core/framework folder is complete stub wise. But the stubs overall are still very work in progress and will not recognize other core parts. You are welcome to contribute stubs for tensorflow and I'll be happy to review them.

**Reply by weiyshay** [reply]
> I assume they mean [tensorflow.core.framework](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/core/framework). The repository doesn't show it but that folder when installed does have some .py files generated from protobuf files there.

Yes, exactly. I have downloaded the source code and put them in a dir under my project and expect I can go to the definitions on demand.

> 
> Tensorflow is very much untyped library though. You can pip install types-tensorflow to get some stubs. They are maintained here in [typeshed](https://github.com/python/typeshed/tree/main/stubs/tensorflow/tensorflow/core/framework) and I think tensorflow/core/framework folder is complete stub wise. But the stubs overall are still very work in progress and will not recognize other core parts. You are welcome to contribute stubs for tensorflow and I'll be happy to review them.

I am happy to contribute while little knowledge on how the stubs work. Appreciate any links to how to start.

**Reply by erictraut** [reply]
Can you please provide specific repro steps for your problem starting with the creation of a fresh venv and project? I'm still unclear about what you have installed and where.
